code/modules/mob/dead/observer/verbs.dm

```diff
@@ -522,29 +522,37 @@
 /mob/dead/observer/verb/become_hobo()
 	set name = "Become Space Hobo"
 	set category = "Ghost"
-	if(!can_reenter_corpse)
+	if(!(mind && mind.current && can_reenter_corpse) || (mind.current.key && copytext(mind.current.key,1,2)!="@"))
```

Comment:
This is outrageously hacky, I am sure there is a better way

Reply:
Even though this is remarkably ugly I talked to Exxion about it and he prefers this, so I guess it can stand

Reply:
It's just both of the corpse re entering checks stacked from the lines above, the last two stop observers becoming them and are needed.